### PR TITLE
Tag services on update

### PR DESCRIPTION
**Pull request by PettitWesley**
Add tags when updating an existing service.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by PettitWesley** [comment]
Testing:

```
$ ecs-cli compose --file justlaunchit.yml  service up --cluster TEST --region us-west-1 --tags its=friday
INFO[0000] Using ECS task definition                     TaskDefinition="compose3:4"
INFO[0000] Updated the ECS service with a new task definition. Old containers will be stopped automatically, and replaced with new ones  desiredCount=1 force-deployment=false service=compose3
INFO[0000] Service status                                desiredCount=1 runningCount=1 serviceName=compose3
INFO[0106] Service status                                desiredCount=1 runningCount=0 serviceName=compose3
INFO[0106] (service compose3) has stopped 1 running tasks: (task 251341fdfa8143a7966014daa0f82fa4).  timestamp="2019-02-23 01:46:10 +0000 UTC"
INFO[0122] Service status                                desiredCount=1 runningCount=1 serviceName=compose3
INFO[0122] (service compose3) has started 1 tasks: (task 711ab2c3af0448f19ef6d1e09cc3ebeb).  timestamp="2019-02-23 01:46:20 +0000 UTC"
INFO[0122] ECS Service has reached a stable state        desiredCount=1 runningCount=1 serviceName=compose3
```

Note that a new revision of the task def was created, this is because the CLI detected that there were new tags.

```
$ aws resourcegroupstaggingapi get-resources --tag-filters Key=its,Values=friday --region us-west-1
{
    "ResourceTagMappingList": [
        {
            "ResourceARN": "arn:aws:ecs:us-west-1:144718711470:service/TEST/compose3",
            "Tags": [
                {
                    "Key": "its",
                    "Value": "friday"
                }
            ]
        },
        {
            "ResourceARN": "arn:aws:ecs:us-west-1:144718711470:task-definition/compose3:4",
            "Tags": [
                {
                    "Key": "its",
                    "Value": "friday"
                }
            ]
        }
    ]
}
```

The service and task definition have the new tags. There are no tasks which have these tags because I created the service without tags, which means its `propogateTags` value was set to `none`.

File: ecs-cli/modules/cli/compose/entity/service/service.go [hunk 1]
```diff
@@ -264,6 +266,47 @@ func (s *Service) Up() error {
...
+			return err
+		}
+		log.Info("Added new tags to service")
+	}
+
+	return nil
+}
+
+// Allows us to inject the mock tagging client in tests
+// This isn't the best design practice, but its convenient since we only need
```
Comment (allisaurus) [inline ecs-cli/modules/cli/compose/entity/service/service.go]: you can probably omit this and the follow line of commentary :P (but leave the TODO)

File: ecs-cli/modules/utils/utils.go [hunk 2]
```diff
@@ -78,3 +78,31 @@ func ParseTags(flagValue string, tags []*ecs.Tag) ([]*ecs.Tag, error) {
 	}
 	return tags, nil
 }
+
+// GetTags parses AWS Resource tags from the flag value
+// users specify tags in this format: key1=value1,key2=value2,key3=value3
+// Returns tags in the format used by the standalone resource tagging API
+func GetTagsMap(flagValue string) (map[string]*string, error) {
```
Comment (allisaurus) [inline ecs-cli/modules/utils/utils.go]: Don't these functions already exist? Is there a rebase needed?

File: ecs-cli/modules/cli/compose/entity/service/service.go [hunk 3]
```diff
@@ -264,6 +266,47 @@ func (s *Service) Up() error {
...
+// tagging client in one place right now
+// TODO: Refactor this if we use the stand alone tagging client in more places in the future
```
Comment (SoManyHs) [inline ecs-cli/modules/cli/compose/entity/service/service.go]: 👍

File: ecs-cli/modules/cli/compose/entity/service/service.go [hunk 4]
```diff
@@ -264,6 +266,47 @@ func (s *Service) Up() error {
...
+			ecsService.ServiceArn,
+		},
+		Tags: tags,
+	}
+	output, err := taggingClient.TagResources(input)
+	if err != nil {
+		return err
+	}
+
+	for resource, info := range output.FailedResourcesMap {
```
Comment (allisaurus) [inline ecs-cli/modules/cli/compose/entity/service/service.go]: can we add a test for this error case?

**Pull request merged**